shared/src/permissions.rs

```diff
@@ -0,0 +1,70 @@
+use std::collections::HashMap;
+
+use once_cell::sync::OnceCell;
+use serde::{Deserialize, Serialize};
+use strum::IntoEnumIterator;
+
+#[derive(
+    Clone,
+    Copy,
+    Hash,
+    Eq,
+    PartialEq,
+    Debug,
+    Serialize,
+    Deserialize,
+    strum_macros::EnumString,
+    strum_macros::ToString,
+    strum_macros::EnumIter,
+)]
+pub enum Permission {
+    AccountPermanentBan,
+    AccountTemporaryBan,
+    AccountList,
+    AccountView,
+    UniverseList,
+    UniverseEdit,
+}
+
+#[derive(
+    Hash,
+    Eq,
+    PartialEq,
+    Debug,
+    Serialize,
+    Deserialize,
+    strum_macros::EnumString,
+    strum_macros::ToString,
+)]
+pub enum Service {
+    Account,
+    Universe,
+}
```

Comment:
```rust
enum Service {
    Account(AccountPermissions),
    Universe(UniversePermissions),
}
```
I think this would use the strength's of Rust even more.

Reply:
The main issue with this is how to serialize it into the db. I chose to use strum to get from_str and to_string, and I could switch to some serde-based encoding instead, but I don't know of a simple way to change this and still keep serialization this lightweight. Serde feels like overkill.